xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/dashboard/dashboard.js

```diff
@@ -75,7 +80,12 @@ XWiki.Dashboard = Class.create( {
     // find out all the gadget-containers in element and add them ids
     this.containers = element.select(".gadget-container");
     this.createDragAndDrops();
-    this.addGadgetsHandlers();
+    this.addGadgetsHandlers();``
+    // Create the section to contain add buttons
+    var sectionAddButtons = new Element('section', {
+      'class': 'containeradd'
+    })
+    this.element.insert({'top' : sectionAddButtons});
```

Comment:
You couldn't actually put that directly in the velocity template? From what I understand this DOM element would always be there no?

Reply:
From what I understand, tehre is no velocity template for the edit part of the dashboard. Everything added to make it nice to edit the dashboard is done so through this .js script.
Those elements are only there in edit mode.
Most of the dashboard is handled in [its own macro](https://github.com/xwiki/xwiki-platform/blob/9a4e759f3189b2125a5d13af6ab32ce24515f5f9/xwiki-platform-core/xwiki-platform-dashboard/xwiki-platform-dashboard-macro/src/main/java/org/xwiki/rendering/internal/macro/dashboard/DashboardMacro.java#L279).